src/ui/router/ApplicationRouterSpec.js

```diff
@@ -54,86 +49,60 @@ xdescribe('Application router utility functions', () => {
     });
 
     it('The setSearchParam function sets an individual search parameter in the window location hash', (done) => {
-        let success;
-        openmct.router.setSearchParam('testParam', 'testValue');
         resolveFunction = () => {
-            success = window.location.hash.includes('testParam=testValue');
-            if (success) {
-                expect(success).toBe(true);
+            openmct.router.setSearchParam('testParam', 'testValue');
+            const searchParams = openmct.router.getAllSearchParams();
+            expect(searchParams.get('testParam')).toBe('testValue');
 
-                openmct.router.removeListener('change:hash', resolveFunction);
-                done();
-            }
+            openmct.router.removeListener('change:hash', resolveFunction);
+            done();
         };
 
         openmct.router.on('change:hash', resolveFunction);
```

Comment:
I don't understand how this test works now, unless `setSearchParam` is synchronous? Here are the steps it is performing.
1. Wait for the hash to change
2. Set the search paramter
3. Verify that the search parameter has been set.

So why are we waiting for that initial hash change?